### PR TITLE
feat(csharp): add compression

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

🎟 JIRA Ticket: DI-1731

### Changes included:
- Add GZip Compression implementation

### CODE REVIEW

File: tests/output/csharp/src/tests/CompressionTests.cs
Comment (millotp) [inline tests/output/csharp/src/tests/CompressionTests.cs:1]: this folder is not meant for hand written test, maybe this should be in the client folder directly, like we do for js

**Review by millotp — COMMENTED** [review]
looks great !

File: tests/output/csharp/src/tests/CompressionTests.cs
Comment (morganleroi) [reply tests/output/csharp/src/tests/CompressionTests.cs:1]: In dotnet we separate tests and production code in two different projects. Maybe I can create a separate folder for theses tests, but inside the tests assembly ? WDYT ?

Comment (millotp) [reply tests/output/csharp/src/tests/CompressionTests.cs:1]: yes, or maybe just outside of src here

Comment (morganleroi) [reply tests/output/csharp/src/tests/CompressionTests.cs:1]: done

**Review by millotp — DISMISSED** [review]

**Review by Fluf22 — APPROVED** [review]
Let's go!

**Pull request merged**